nav2_amcl/src/amcl_node.cpp

```diff
@@ -845,18 +845,18 @@ AmclNode::laserReceived(sensor_msgs::msg::LaserScan::ConstSharedPtr laser_scan)
       // p.covariance[6*5+5] = hyps[max_weight_hyp].pf_pose_cov.m[2][2];
       p.pose.covariance[6 * 5 + 5] = set->cov.m[2][2];
 
-      /*
-         printf("cov:\n");
-         for(int i=0; i<6; i++)
-         {
-         for(int j=0; j<6; j++)
-         printf("%6.3f ", p.covariance[6*i+j]);
-         puts("");
-         }
-       */
+      float temp = 0.0;
+      for (int i = 0; i < static_cast<int>(p.pose.covariance.size()); i++) {
```

Comment:
I assume you're doing the static_cast<int> to silence the unsigned/signed mismatch warning? Why not just make i unsigned?

Better yet, just use the new range based for loop :

```
for( auto covariance_value : p.pose.covariance) {
  temp+= covariance_value;
}
```